src/libraries/System.Private.Xml/src/System/Xml/Dom/DocumentXPathNavigator.cs

```diff
@@ -1437,6 +1451,8 @@ public override XmlNodeOrder ComparePosition(XPathNavigator other)
                 {
                     return XmlNodeOrder.After;
                 }
+                Debug.Assert(node1 != null);
+                // TODO: What ensures that node1 is not null.
```

Comment:
honestly, I gave up with full analysis on everything and just add `!` - I'd only do better analysis on stuff affecting public APIs - for cases where you have multiple implementations of XPathNavigator like here I'd go with whatever majority does (unless docs have documented expected behavior)